File: backend/api/schemas/task.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel, UUID4
from api.models.task import StatusTarefa, PrioridadeTarefa

class TaskCreate(BaseModel):
    titulo: str
    descricao: str
    status: StatusTarefa
    prioridade: PrioridadeTarefa
    data_vencimento: datetime
    usuario_id: Optional[UUID4] = None
    
    class Config:
        from_attributes = True

class TaskUpdate(BaseModel):
    titulo: Optional[str] = None
    descricao: Optional[str] = None
    status: Optional[StatusTarefa] = None
    prioridade: Optional[PrioridadeTarefa] = None
    data_vencimento: Optional[datetime] = None
    usuario_id: Optional[UUID4] = None
    
    class Config:
        from_attributes = True

class TaskRead(BaseModel):
    id: UUID4
    titulo: str
    descricao: str
    status: StatusTarefa
    prioridade: PrioridadeTarefa
    data_vencimento: datetime
    created_at: datetime
    updated_at: datetime
    usuario_id: UUID4

File: frontend/lib/types/task.ts
export type TaskStatus = 'PENDENTE' | 'EM_ANDAMENTO' | 'CONCLUIDA'
export type TaskPriority = 'BAIXA' | 'MEDIA' | 'ALTA'

export interface Task {
  id: string
  titulo: string
  descricao: string
  status: TaskStatus
  prioridade: TaskPriority
  data_vencimento: string | undefined
  created_at: string
  updated_at: string
  usuario_id: string,
  ordem?: number
}

export interface TaskCreate {
  titulo: string
  descricao: string
  status: TaskStatus
  prioridade: TaskPriority
  data_vencimento: string | undefined
  ordem?: number
}

export interface TaskUpdate {
  titulo?: string
  descricao?: string
  status?: TaskStatus
  prioridade?: TaskPriority
  data_vencimento?: string | undefined
  ordem?: number
}

export interface PaginatedResponse<T> {
  total: number
  data: T[]
}

export interface SingleResponse<T> {
  data: T[]
}

export interface TaskFilters {
  status?: TaskStatus
  prioridade?: TaskPriority
  search?: string
  skip?: number
  limit?: number
  data_criacao?: string
  data_atualizacao?: string
  data_vencimento?: string
} 